Content.Server/Store/Conditions/StoreAntagCondition.cs

```diff
@@ -0,0 +1,63 @@
+using Content.Server.Mind.Components;
+using Content.Server.Traitor;
+using Content.Shared.Roles;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+
+namespace Content.Shared.Store.Conditions;
+
+/// <summary>
+/// Allows a store entry to be filtered out based on the user's antag role.
+/// Supports both blacklists and whitelists. This is copypaste because roles
+/// are absolute shitcode. Refactor this later. -emo
+/// </summary>
+public sealed class StoreAntagCondition : ListingCondition
```

Comment:
stuff like supersurplus being attached to `StoreAntagCondition` IMO doesn't make a lot of sense. this means that if a non-nukie gets ahold of a syndicate radio, they can use it to order a super surplus crate/holopara because they don't actually have the antag role. i really feel like it should be just similar to `StoreUserWhitelist` but for the uplink entity, and then the radio can have a special tag that blacklists nukie items